src/misc/multi-env-map.js

```diff
@@ -0,0 +1,147 @@
+/**
+ * @param  {Array<THREE.Material>|THREE.Material} material
+ * @return {Array<THREE.Material>}
+ */
+function ensureMaterialArray (material) {
+  if (!material) {
+    return [];
+  } else if (Array.isArray(material)) {
+    return material;
+  } else if (material.materials) {
+    return material.materials;
+  } else {
+    return [material];
+  }
+}
```

Comment:
I'm only aware of two possibilities here, and would expect this to be enough:

```js
return Array.isArray( material ) ? material : [ material ];
```

Do you know of situations where a mesh might not have a material, or where `material.materials` might come from?